services/src/test/java/org/jfrog/artifactory/client/ArtifactoryTestsBase.java

```
@@ -208,6 +208,32 @@ protected String textFrom(InputStream is) throws IOException {
         }
     }
 
+    protected void deleteRepoWithRetry(String repoKey) {
```

PR looks quite on point, well coded and well commented wherever needed!!
**Good Parts:**
- The retry logic is clear and targeted at a specific transient error.
- The sleep and interrupt handling are proper.
- The method avoids retrying for non-lock errors.

**Possible Improvements:** (if possible and if needed)
- **Logging:** There is no logging in the catch block. Adding a log statement for retries and errors can help debugging and traceability.

something like,
```
// Log retry attempt
    System.err.println("Attempt " + attempt + " failed due to lock; retrying...");
// Log or throw after final failure
    System.err.println("Failed to delete repo " + repoKey + " after " + MAX_RETRIES + " attempts.");
```

- **Error Handling:** If all retries fail due to lock, the method silently returns. Consider throwing an exception or logging a warning after exhausting retries,

- You mentions retries and delay like this on top:
```
private static final int MAX_RETRIES = 3;
private static final int RETRY_DELAY_MS = 5000;
```

---

thank you for the input. I will incorporate them.